public/pages/utils/anomalyResultUtils.ts

```diff
@@ -1179,3 +1253,164 @@ export const parseEntityAnomalySummaryResults = (
   } as EntityAnomalySummaries;
   return enityAnomalySummaries;
 };
+
+export const getAnomalyDataRangeQuery = (
+  startTime: number,
+  endTime: number,
+  taskId: string
+) => {
+  return {
+    size: 0,
+    query: {
+      bool: {
+        filter: [
+          {
+            range: {
+              [AD_DOC_FIELDS.ANOMALY_GRADE]: {
+                gte: 0,
```

Comment:
We use `gte` here since we want to include all results (anomalous or not) to get the range of existing results. It's just an optimization to shrink down the search space for the following query. We use `gt` in `getHistoricalAggQuery` since we only care about aggregating with anomalous results (max grade is the only agg currently supported).

Reply:
still don't get it.  In AnomalyDetailsChart, getAnomalyDataRangeQuery's output is the input of getHistoricalAggQuery.  So if getHistoricalAggQuery only tries to get max non-zero grade in buckets, why  getAnomalyDataRangeQuery wants to give min-max range of all results? If getAnomalyDataRangeQuery only gives time range of non-zero grade, would that change the return result of getHistoricalAggQuery?

Reply:
The idea is that `getAnomalyDataRangeQuery`'s output is just shrinking down the search space needed for `getHistoricalAggQuery`. We don't care about aggregating time buckets that are guaranteed to be empty, so the first query is just providing a beginning and end time range, where there is guaranteed to be data points (anomalous or not) within that range. In the context of [this comment](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/16#discussion_r666547249), it is basically trimming off the `x`s.

Reply:
> If getAnomalyDataRangeQuery only gives time range of non-zero grade, would that change the return result of getHistoricalAggQuery?

Yes, it would, because it would prevent us from aggregating over existing, non-anomalous data points, which we don't want to do. For example, given some time bucket that has 5 data points all with `anomaly_grade` 0, we still want to aggregate and display the `max` `anomaly_grade` as 0, rather than having no data point at all.

Reply:
Since you filtered out zero grade results and do a max aggregation, you would end up with null instead of 0 since max of missing data (missing because you filtered it out) is null.  Does that matter?

Reply:
Correct. Like you mentioned in [this comment](https://github.com/opensearch-project/anomaly-detection-dashboards-plugin/pull/16#discussion_r667168386), we will set null max grades to 0. By filtering out zero grade results here it optimizes the query since far fewer results need to be processed & aggregated.